### PR TITLE
BREAKING CHANGE: Avoid collisions between static class/namespace members

### LINKED ISSUE #2793: asc crash when use same name in class and namespace
### Bug description

```ts
class NS {
  static v: i32 = 2;
}

namespace NS {
  let v: i32 = 1;
}
```

```
Fatal: Module::addGlobal: assembly/index/NS.v already exists

▌ Whoops, the AssemblyScript compiler has crashed during compile :-(
▌ 
▌ There is no stack trace. Perhaps a Binaryen exception above / in console?
▌ 
▌ > undefined
▌ 
▌ If you see where the error is, feel free to send us a pull request. If not,
▌ please let us know: https://github.com/AssemblyScript/assemblyscript/issues
▌ 
▌ Thank you!
```

### Steps to reproduce

`npm run asbuild`

### AssemblyScript version

latest

### PR DESCRIPTION
**Pull request by CountBleck**
Fixes #2793.

Changes proposed in this pull request:
⯈ Change the delimiter between namespaces and their members from `.` to `::`

- [x] I've read the contributing guidelines
- [x] I've added my name and email to the NOTICE file

### CODE REVIEW

**Review by HerrCai0907 — COMMENTED** [review]
should we do diagnose first before merge this PR?
```
class NS {
  static v: i32 = 2;
}

namespace NS {
  export let v: i32 = 1;
}
```
[TS playground](https://www.typescriptlang.org/play?#code/MYGwhgzhAEByDK0DeAoa0IBcyYJbGgDcAuaAOwFcBbAIwFMAnaAXmgCYBuFAXxRTLBU6EAA5hgdOIlTo6ADxEB7Bpmgg6qkuWr0mrAIxdeKYIrIRF6gHQhFAcwAUCK4QCUQA)

**Comment by CountBleck** [comment]
> should we do diagnose first before merge this PR?

Actually, this makes me realize how redundant this PR is, since if that diagnostic is emitted, then there wouldn't be a collision.

**Comment by CountBleck** [comment]
Never mind, the diagnostic is only needed for exported namespace members and public static class members.

File: src/common.ts [hunk 1]
```diff
@@ -94,10 +94,18 @@ export const PARENT_SUBST = "..";
...
+ */
 export const INSTANCE_DELIMITER = "#";
-/** Delimiter used between class and namespace names and static members. */
+/**
+ * Delimiter used between class names and static members, as well as enum
+ * names and enum members.
+ */
 export const STATIC_DELIMITER = ".";
+/** Delimiter used between namespace names and namespace members. */
+export const NAMESPACE_DELIMITER = "::";
```
Comment (MaxGraey) [inline src/common.ts:108]: What about make it shorter and use only one symbol?
```suggestion
export const NAMESPACE_DELIMITER = ":";
```

**Comment by CountBleck** [comment]
It looks like supporting this properly is a pain, since modifying @HerrCai0907's example to remove the `export` still wouldn't work with this PR because of how declarations are merged. Accesses to `NS.v` end up targeting the namespace's `v` member, whether or not that member is exported.

It's probably best to diagnose this in every case.

**Pull request closed without merging**